pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -26,21 +33,34 @@ func ProvideService(sqlStore sqlstore.Store, secretsService secrets.Service, rem
 		},
 	}
 	if remoteCheck.ShouldUseRemoteSecretsPlugin() {
-		logger.Debug("secrets kvstore is using a remote plugin for secrets management")
-		secretsPlugin, err := remoteCheck.GetPlugin()
-		if err != nil {
-			logger.Error("plugin client was nil, falling back to SQL implementation")
+		// Attempt to start the plugin
+		secretsPlugin, err := remoteCheck.StartAndReturnPlugin(context.TODO())
```

Comment:
I believe we can use `context.Background()` in this case since that's what the plugin manager uses to load plugins on https://github.com/grafana/grafana/tree/main/pkg/plugins/manager/manager.go#L69

Same goes for the other `context.TODO()`.

Reply:
done